tabular/src/autogluon/tabular/trainer/abstract_trainer.py

```diff
@@ -364,6 +367,12 @@ def stack_new_level_aux(self, X, y, base_model_names: List[str], level, fit=True
         Auxiliary models never use the original features and only train with the predictions of other models as features.
         """
         X_train_stack_preds = self.get_inputs_to_stacker(X, base_models=base_model_names, fit=fit, use_orig_features=False)
+        if self.weight_evaluation:
+            X, w = extract_column(X, self.weight_column)
+            if w is not None:
+                sample_weights = w.values/w.sum() * len(w)
+                X_train_stack_preds[self.weight_column] = sample_weights
```

Comment:
I don't think weight column should impact this function. Shouldn't be necessary if w is its own variable.

Reply:
Added TODO to consider not bundling weights inside of X in future internal refactor